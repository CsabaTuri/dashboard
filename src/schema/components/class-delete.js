import React from "react"
import { connect } from "react-redux"
import { browserHistory } from "react-router"
import ReactGA from "react-ga"

import DeleteForm from "./delete-form"
import { deleteClass, selectedDatabase, selectedClass } from "../"
import { notify } from "../../notifications"
import { faunaClient } from "../../authentication"
import { buildResourceUrl } from "../../router"

export const ClassDelete = ({ client, path, dbUrl, clazz }) => {
  const onDelete = () => {
    ReactGA.event({category: "schema", action: "class-create"});
    return notify("Class deleted successfully", dispatch =>
      dispatch(deleteClass(client, path, clazz.get("name"))).then(() =>
        browserHistory.push(buildResourceUrl(dbUrl, "classes"))
      )
    )
  }

  return <DeleteForm
    buttonText="Delete Class"
    type="class"
    title={`Delete ${clazz.get("name")}`}
    validateName={clazz.get("name")}
    onDelete={onDelete} />
}

ClassDelete.displayName = "ClassDelete"

export default connect(
  state => ({
    client: faunaClient(state),
    path: selectedDatabase(state).get("path"),
    dbUrl: selectedDatabase(state).get("url"),
    clazz: selectedClass(state)
  })
)(ClassDelete)
