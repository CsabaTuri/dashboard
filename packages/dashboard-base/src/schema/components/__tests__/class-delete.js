import React from "react"
import { Map } from "immutable"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import { ClassDelete } from "../class-delete"

jest.mock("../../../notifications", () => ({
  notify: (msg, fn) => fn(() => Promise.resolve())
}))

jest.mock("react-router", () => ({
  browserHistory: { push: jest.fn() }
}))

const { browserHistory } = require("react-router")

describe("ClassDelete Component", () => {
  let comp, store, client

  beforeEach(() => {
    comp = shallow(
      <ClassDelete
        client={client}
        path={["a-db"]}
        clazz={Map.of("name", "fake-class")}
        dbUrl="/db/a-db/databases" />
    )
  })

  it("renders delete form", () => {
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("delete class", () => {
    return comp.props().onDelete().then(() => {
      expect(browserHistory.push).toHaveBeenCalledWith("/db/a-db/classes")
    })
  })
})

