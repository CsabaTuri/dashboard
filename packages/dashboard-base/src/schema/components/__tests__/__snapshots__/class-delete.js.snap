// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassDelete Component renders delete form 1`] = `
<Connect(DeleteForm)
  buttonText="Delete Class"
  onDelete={[Function]}
  title="Delete fake-class"
  type="class"
  validateName="fake-class"
/>
`;
